### PR TITLE
 Refactor led-raspberri-pi demo to make use of device CRDs

### LINKED ISSUE #17: Update led-raspberrypi demo to make use of device CRD
Update the code in led-raspberrypi demo to make use of device CRD feature of kubeedge

### PR DESCRIPTION
**Pull request by sujithsimon22**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**
This PR modifies the led-raspberri-pi demo by refactoring the led-light-mapper to make use of device CRDs, which is a major feature of kubeedge v03

**Which issue(s) this PR fixes**
Fixes #17

### CODE REVIEW

**Comment by sujithsimon22** [comment]
cc @rohitsardesai83 @sids-b @m1093782566  for review

File: led-raspberrypi/README.md [hunk 1]
```diff
@@ -32,53 +32,61 @@ When power is provided in the pin, the LED glows (ON State) and when no power is
...
 ## Steps to reproduce
 
 1. Connect the LED to the RaspBerry-Pi using the GPIO as shown in the [circuit diagram](images/raspberry-pi-wiring.png) above.   
 
 2. Clone and run KubeEdge. 
-    Please click [KubeEdge Usage](https://github.com/kubeedge/kubeedge#usage) for instructions on the usage of KubeEdge.
-Note: Please start KueEdge with in internal or double MQTT mode.
+    Please click [KubeEdge Usage](https://github.com/kubeedge/kubeedge/blob/master/docs/getting-started/usage.md) for instructions on the usage of KubeEdge.
+
+3. Clone the led_raspberrypi_demo
```
[the remark this answers is not in the document]
Comment (sujithsimon22) [reply led-raspberrypi/README.md]: done, thanks

File: led-raspberrypi/README.md [hunk 2]
```diff
@@ -32,53 +32,61 @@ When power is provided in the pin, the LED glows (ON State) and when no power is
...
+2. KubeEdge (Version 0.3 has been used for this demo)
 
 ## Steps to reproduce
 
 1. Connect the LED to the RaspBerry-Pi using the GPIO as shown in the [circuit diagram](images/raspberry-pi-wiring.png) above.   
 
 2. Clone and run KubeEdge. 
-    Please click [KubeEdge Usage](https://github.com/kubeedge/kubeedge#usage) for instructions on the usage of KubeEdge.
-Note: Please start KueEdge with in internal or double MQTT mode.
+    Please click [KubeEdge Usage](https://github.com/kubeedge/kubeedge/blob/master/docs/getting-started/usage.md) for instructions on the usage of KubeEdge.
```
[the remark this answers is not in the document]
Comment (sujithsimon22) [reply led-raspberrypi/README.md]: done, thanks

File: led-raspberrypi/README.md [hunk 3]
```diff
@@ -32,53 +32,61 @@ When power is provided in the pin, the LED glows (ON State) and when no power is
...
-4.  Clone the led_raspberrypi_demo 
+       ```shell
+                   cd $GOPATH/src/github.com/kubeedge/examples/led-raspberrypi/sample-crds
+                   kubectl apply -f led-light-device-model.yaml
+                   kubectl apply -f led-light-device-instance.yaml
+
+                   Note: You can change the CRDs to match your requirement
+        ```
+
+ 5. Update the name of the device (device instance name) created using the device CRD in the previous step along with the MQTT mode using which edge_core is running in the configuration file present at $GOPATH/src/github.com/kubeedge/examples/led-raspberrypi/configuration/config.yaml
```
[the remark this answers is not in the document]
Comment (sujithsimon22) [reply led-raspberrypi/README.md]: that is a good point. I have changed it as requested

File: led-raspberrypi/README.md [hunk 4]
```diff
@@ -32,53 +32,61 @@ When power is provided in the pin, the LED glows (ON State) and when no power is
...
+    Please click [KubeEdge Usage](https://github.com/kubeedge/kubeedge/blob/master/docs/getting-started/usage.md) for instructions on the usage of KubeEdge.
+
+3. Clone the led_raspberrypi_demo
+
+       ```shell
+                   git clone https://github.com/kubeedge/examples.git $GOPATH/src/github.com/kubeedge/examples
+        ```
 
-3. Create the LED device in the cloud, with a device twin attribute called "Power Status" (This is used to turn the device ON/OFF). 
+4. Create the LED device in the cloud, by creating device model and device instance CRDs
```
[the remark this answers is not in the document]
Comment (sujithsimon22) [reply led-raspberrypi/README.md]: done, thanks

**Comment by sujithsimon22** [comment]
> There is a synctocloud function, is it required ? I think e just need to update the twin status and deviceTwin will ensure sending update message to cloud after update

I have removed synctocloud function, thanks

**Comment by sujithsimon22** [comment]
cc @sids-b for review

**Comment by m1093782566** [comment]
/lgtm

@sids-b Please feel free to merge.

**Pull request merged**